librosa/beat.py

```diff
@@ -374,7 +374,7 @@ def tempo(
 
     # Get the maximum, weighted by the prior
     # Using log1p here for numerical stability
-    best_period = np.argmax(np.log1p(1e6 * tg) + logprior[:, np.newaxis], axis=0)
+    best_period = np.argmax(np.log1p(1e6 * tg) + logprior[:, np.newaxis], axis=-2)
 
     return bpms[best_period]
```

Comment:
Done, but not clear to me whether you mean `best_period` may be a multi-element vector (instead of just a list with one scalar), or something with more dimensions (i.e. matrix or ndarray).

Reply:
> or something with more dimensions (i.e. matrix or ndarray).

Yes: if you do `np.argmax(X, axis=n)`, the result will have shape equal to `X.shape` except dimension `n` is removed.  

In our case, we used to have `tg.ndim=2`, so argmaxing results in a 1d array, so `best_period` has shape `(n_frames,)`.  When we expand out to multichannel, we'll be collapsing out the tempo dimension, but retaining channels and frames.

My comment here comes from the concern that using a multidimensional index array can sometimes do unexpected things (see [here](https://numpy.org/doc/stable/reference/arrays.indexing.html#advanced-indexing)).  However, in this case we're okay because `bpms` is always 1d, so we don't get into any of the weird cases.